lib/Sugar.ml

```diff
@@ -439,25 +445,7 @@ module Let_binding = struct
     let is_local_pattern, lb_pat, lb_exp =
       maybe_sugar_local cmts ~ctx pvb_pat pvb_expr pvb_is_pun pvb_constraint
     in
-    let ({ast= pat; _} as xpat) =
-      match (lb_pat.ast.ppat_desc, lb_exp.ast.pexp_desc) with
-      (* recognize and undo the pattern of code introduced by
-         ocaml/ocaml@fd0dc6a0fbf73323c37a73ea7e8ffc150059d6ff to fix
-         https://caml.inria.fr/mantis/view.php?id=7344 *)
```

Comment:
This desugaring no longer happens; the constraint is now encoded in `pvb_constraint`.